detectable/src/main/java/com/synopsys/integration/detectable/detectables/docker/DockerExtractor.java

```diff
@@ -193,4 +202,21 @@ private Extraction executeDocker(final File outputDirectory, final String imageA
         return new Extraction.Builder().failure("No files found matching pattern [" + DEPENDENCIES_PATTERN + "]. Expected docker-inspector to produce file in " + directory.toString());
     }
 
+    private String getImagePieceFromOutputDirectoryIfImageIdPresent(File outputDirectory, String imageArgument, String imagePiece) {
+        if (imageArgument.contains("image.id")) {
+            final File producedResultFile = fileFinder.findFile(outputDirectory, RESULTS_FILENAME_PATTERN);
+            String jsonText = null;
+            try {
+                jsonText = FileUtils.readFileToString(producedResultFile, StandardCharsets.UTF_8);
+            } catch (final IOException e) {
+                e.printStackTrace();
+            }
+            DockerImageInfo dockerImageInfo = gson.fromJson(jsonText, DockerImageInfo.class);
+
+            return dockerImageInfo.getImageRepo() + ":" + dockerImageInfo.getImageTag();
+        } else {
+            return imagePiece;
+        }
+    }
+
```

Comment:
I didn't think of this originally, but I think this method should gracefully handle failure to get those values. Let's assume the file might not be there (which will be true for old versions of Docker Inspector). Let's also assume dockerImageInfo.getImageRepo() might return null or "". While we're at it, let's assume gson.fromJson() might fail or return nothing. In any of those cases, let's fall back to using the image ID in the codelocation name, so Detect succeeds and the user gets results.

Reply:
Also we shouldn't just print a stack trace. Worse case it should log at debug level. Something like: 

`logger.debug("Could not find image id", e);`